samza-sql/src/main/java/org/apache/samza/sql/translator/ProjectTranslator.java

```diff
@@ -53,28 +60,42 @@
     this.queryId = queryId;
   }
 
-  private static class ProjectMapFunction implements MapFunction<SamzaSqlRelMessage, SamzaSqlRelMessage> {
+  @VisibleForTesting
+  public static class ProjectMapFunction implements MapFunction<SamzaSqlRelMessage, SamzaSqlRelMessage> {
     private transient Project project;
     private transient Expression expr;
     private transient TranslatorContext context;
+    private transient MetricsRegistry metricsRegistry;
+    private transient SamzaHistogram processingTimeHistogram; // milli-seconds
+    private transient Counter messagesCounter;
 
     private final int queryId;
     private final int projectId;
+    private final String logicalOpId;
+    private final String PROCESSING_TIME_HISTOGRAM_NAME = "processingTimeHistogram";
+    private final String MESSAGES_COUNTER_NAME = "messagesCounter";
 
-    ProjectMapFunction(int projectId, int queryId) {
+    ProjectMapFunction(int projectId, int queryId, String logicalOpId) {
```

Comment:
We should have added doc on this method before. Looks like we missed it. Can you add it please?

Reply:
done